exchange/utils.go

```diff
@@ -137,20 +137,16 @@ func splitBidRequest(req *openrtb.BidRequest,
 		return nil, []error{err}
 	}
 
-	var requestExt openrtb_ext.ExtRequest
 	var sChainsByBidder map[string]*openrtb_ext.ExtRequestPrebidSChainSChain
-	if len(req.Ext) > 0 {
-		err := json.Unmarshal(req.Ext, &requestExt)
-		if err != nil {
-			return nil, []error{err}
-		}
 
-		sChainsByBidder, err = BidderToPrebidSChains(&requestExt)
-		if err != nil {
-			return nil, []error{err}
-		}
-	} else {
-		sChainsByBidder = make(map[string]*openrtb_ext.ExtRequestPrebidSChainSChain)
+	sChainsByBidder, err = BidderToPrebidSChains(requestExt)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	reqExt, err := prepareExt(req, requestExt)
+	if err != nil {
+		return nil, []error{err}
 	}
```

Comment:
> Did you change the signature of prepareExt so you could early return?

Yes.

> I was aiming for simple prepare method signatures with similar behavior

Good call. I'll change the name to `parseExt` since those functions do return errors.